source/AI.cpp

```diff
@@ -522,8 +522,11 @@ void AI::Step(const PlayerInfo &player, Command &activeCommands)
 		}
 
 	const Ship *flagship = player.Flagship();
-	step = (step + 1) & 31;
+	// Yes, this overflows. but the beauty of bitwise-and means that
+	// overflow doesn't cause any issues.
+	step = step + 1;
```

Comment:
Overflow _can_ cause issues for signed integers such as `step`.

Reply:
step is only used as part of bitwise-and. (it's only used on line 635 and 558). Bitwise-and, unlike modulus, will always return a positive number if at least one of the arguments is positive.
In other words, the way step is used with bitwise-and means that only the less significant bits matter, and they aren't effected by overflow.

Reply:
Signed integer overflow is undefined behavior. I doubt that this will cause issues in this specific instance, but still. Why not do this:

```suggestion
	step = (step + 1) & 127;
```